core/src/display_object/movie_clip.rs

```diff
@@ -4649,12 +4689,35 @@ impl<'gc> MovieClipStatic<'gc> {
         swf: SwfSlice,
         total_frames: FrameNumber,
         loader_info: Option<Avm2Object<'gc>>,
+        is_root: bool,
         gc_context: &Mutation<'gc>,
     ) -> Self {
+        let mut show_frame_count = 0;
+        let tag_callback = |_reader: &mut SwfStream<'_>, tag_code, _tag_len| {
+            match tag_code {
+                TagCode::ShowFrame => {
+                    show_frame_count += 1;
+                }
+                TagCode::End => {
+                    return Ok(ControlFlow::Exit);
+                }
+                _ => {}
+            };
+            Ok(ControlFlow::Continue)
+        };
+
+        if is_root && !swf.is_empty() {
+            let mut reader = swf.read_from(0);
+            let _ = tag_utils::decode_tags(&mut reader, tag_callback);
```

Comment:
Does this mean we have to decode all the SWF tags twice now?

Reply:
Yes, but the actual decoding work here is just repeatedly reading tag codes and lengths. Also, it's only done for root clips.